### PR TITLE
Insights/repo itr metrics

### PR DESCRIPTION
**Pull request by coury-clark**
Adds a counter to the repository iterator used in both the commit indexer and the historical backfiller that will report on each repo with a label for success or failure.

### CODE REVIEW

**Comment by efritz** [comment]
Im reviewing on mobile so I may have missed it,  it it doesn't look like you're registering the Prometheus metric anywhere. This will have the effect of you being able to increment a counter in memory, but it will not expose it in the metric endpoint.

**Comment by coury-clark** [comment]
> Im reviewing on mobile so I may have missed it, it it doesn't look like you're registering the Prometheus metric anywhere. This will have the effect of you being able to increment a counter in memory, but it will not expose it in the metric endpoint.

It seems that using `promauto` vs `prometheus` package will auto register the metric to the default registry. Do we use a different registry? It seemed to work fine on the local Sourcegraph

File: enterprise/internal/insights/compression/worker.go [hunk 1]
```diff
@@ -44,16 +46,18 @@ func NewCommitIndexer(background context.Context, base dbutil.DB, insights dbuti
...
 
-	iterator := discovery.AllReposIterator{
-		IndexableReposLister:  dbcache.NewIndexableReposLister(repoStore),
-		RepoStore:             repoStore,
-		Clock:                 time.Now,
-		SourcegraphDotComMode: envvar.SourcegraphDotComMode(),
+	iterator := discovery.NewAllReposIterator(
+		dbcache.NewIndexableReposLister(repoStore),
+		repoStore,
+		time.Now,
```
Comment (eseliger) [inline enterprise/internal/insights/compression/worker.go:52]: Not related to this diff, but maybe it should use `timeutil.Now`? That one would always use UTC, which is what we do everywhere else in the app.

**Review by eseliger — APPROVED** [review]

File: enterprise/internal/insights/compression/worker.go (hunk 1 above)
Comment (coury-clark) [reply enterprise/internal/insights/compression/worker.go:52]: Cool! Didn't know about that - I'll keep it in mind. Thanks!

**Pull request merged**